x-pack/legacy/plugins/apm/server/lib/transactions/get_transaction/index.ts

```diff
@@ -41,3 +41,44 @@ export async function getTransaction(
   const resp = await client.search<Transaction>(params);
   return resp.hits.hits[0]?._source;
 }
+
+export async function getTransactionByTraceId(traceId: string, setup: Setup) {
```

Comment:
This won't fetch only a root transaction. If only transactions with parent.id are found, one of them will be returned.

Reply:
This should be in a separate file